File: PHP/KidsWithTheGreatestNumberOfCandies.php
<?php
/**
 * @version PHP 8.2.0
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */
namespace PHP;

class KidsWithTheGreatestNumberOfCandies {    

    /**
     * @param Integer[] $candies
     * @param Integer $extraCandies
     * @return Boolean[]
     */
    function kidsWithCandies($candies, $extraCandies) {
        $maximum = 0;
        $output = [];

        foreach ($candies as $candy) {
            $check = ($candy > $maximum) && ($maximum = $candy);
        }

        for ($i = 0; $i < count($candies); $i++) {
            array_push($output, $candies[$i] + $extraCandies >= $maximum);
        }

        return $output;
    }

}

File: GO/KidsWithTheGreatestNumberOfCandies.go
/**
 * @version GO 1.20.3
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 */
package GO

func kidsWithCandies(candies []int, extraCandies int) []bool {
	max := 0
	out := make([]bool, len(candies))

	for _, v := range candies {
		if v > max {
			max = v
		}
	}

	for i, v := range candies {
		if v+extraCandies >= max {
			out[i] = true
		}
	}

	return out
}
